kfs-core/src/main/java/edu/arizona/kfs/gl/businessobject/EntryBo.java

```diff
@@ -0,0 +1,397 @@
+
+        for (AccountingLine accountingLine : accountingLineList) {
+            String documentNumber = accountingLine.getDocumentNumber();
+            DocumentRouteHeaderValue docHeader = getBusinessObjectService().findBySinglePrimaryKey(DocumentRouteHeaderValue.class, documentNumber);
+            if (docHeader != null) {
+                String docTypeName = docHeader.getDocumentType().getName();
+                if (docTypeName.equals(KFSConstants.FinancialDocumentTypeCodes.GENERAL_ERROR_CORRECTION)) {
+                    return documentNumber;
+                }
+            }
+        }
+        return KFSConstants.EMPTY_STRING;
+    }
+
+    private String generateEntryId() {
+        StringBuilder retval = new StringBuilder();
+        retval.append(getUniversityFiscalYear());
+        retval.append(SecConstants.SecurityValueSpecialCharacters.MULTI_VALUE_SEPERATION_CHARACTER);
+        retval.append(getChartOfAccountsCode());
+        retval.append(SecConstants.SecurityValueSpecialCharacters.MULTI_VALUE_SEPERATION_CHARACTER);
+        retval.append(getAccountNumber());
+        retval.append(SecConstants.SecurityValueSpecialCharacters.MULTI_VALUE_SEPERATION_CHARACTER);
+        retval.append(getSubAccountNumber());
+        retval.append(SecConstants.SecurityValueSpecialCharacters.MULTI_VALUE_SEPERATION_CHARACTER);
+        retval.append(getFinancialObjectCode());
+        retval.append(SecConstants.SecurityValueSpecialCharacters.MULTI_VALUE_SEPERATION_CHARACTER);
+        retval.append(getFinancialSubObjectCode());
+        retval.append(SecConstants.SecurityValueSpecialCharacters.MULTI_VALUE_SEPERATION_CHARACTER);
+        retval.append(getFinancialBalanceTypeCode());
+        retval.append(SecConstants.SecurityValueSpecialCharacters.MULTI_VALUE_SEPERATION_CHARACTER);
+        retval.append(getFinancialObjectTypeCode());
+        retval.append(SecConstants.SecurityValueSpecialCharacters.MULTI_VALUE_SEPERATION_CHARACTER);
+        retval.append(getUniversityFiscalPeriodCode());
+        retval.append(SecConstants.SecurityValueSpecialCharacters.MULTI_VALUE_SEPERATION_CHARACTER);
+        retval.append(getFinancialDocumentTypeCode());
+        retval.append(SecConstants.SecurityValueSpecialCharacters.MULTI_VALUE_SEPERATION_CHARACTER);
+        retval.append(getFinancialSystemOriginationCode());
+        retval.append(SecConstants.SecurityValueSpecialCharacters.MULTI_VALUE_SEPERATION_CHARACTER);
+        retval.append(getDocumentNumber());
+        retval.append(SecConstants.SecurityValueSpecialCharacters.MULTI_VALUE_SEPERATION_CHARACTER);
+        retval.append(getTransactionLedgerEntrySequenceNumber());
+        return retval.toString();
+    }
+
+    /**
+     * Generates a Map of fieldValues for a Primary Key lookup based on the given entryId.
+     *
+     * @param entryId
+     * @return
+     */
+    public static Entry getEntry(String entryId) {

```

Comment:
Since this is all about efficiency changes, I think this method should only accept a List of entry IDs, and then ultimately calling the BusinessObjectService method:
public <T extends BusinessObject> Collection<T> findMatching(Class<T> clazz, Map<String, ?> fieldValues);

Looking at GeneralErrorCorrectionAction:199, this is being called in a for loop. We should rewrite so that this is batched up into one call. Any reason you can't?